backend/src/main/java/com/empire/World.java

```diff
@@ -2599,6 +2600,23 @@ public int compare(Node a, Node b) {
 			notifyAll("Harvest Preparations", "People all over the world prepare to reap the harvest upon which they have labored, and hunger is (perhaps only temporarily) banished. All regions will produce their harvest this turn (before eating).");
 		}
 
+		// Check for game end.
+		if (date >= 27 && (date - 27) % 6 == 0) {
+			int votesToEnd = 0;
+			int votesToExtend = 0;
+			for (String kingdom : kingdoms.keySet()) {
+				if ("end".equals(orders.get(kingdom).getOrDefault("end_vote", "end"))) votesToEnd++;
```

Comment:
I've noticed this a lot in the codebase: "literal".equals(valueBeingCompared).  Is this accepted style?  It's somewhat the opposite of how I read and compare things.  Is this one of those 50-50 disagreements like tabs vs. spaces?

Reply:
Agreed this is a yoda statement, but in Java there is a reason for doing so:

Say String s = null.
"foo".equals(s) == false
s.equals("foo") -> NullPointerException.

There's a counterargument from the no-yoda camp that if s is null, returning false is buggy: s should not be null or should be explicitly checked for nullness, and throwing an exception is better in such a case. Personally I find it too verbose to do: s != null && s.equals("foo"), and prefer not to make global data sanity assumptions like "orders will never contain a null value".